src/features/expenses/AddExpense/SplitMethods/Unequally.tsx

```diff
@@ -1,17 +1,78 @@
-import { ICreateTransaction } from "features/expenses/types";
+import { useCallback, useEffect, useRef, useState } from "react";
+import { useDispatch } from "react-redux";
+
+import { Column } from "components/Containers";
+import { ICreateTransaction, IPartialAssignments } from "features/expenses/types";
 import { ITransactionGroup } from "features/groups/types";
+import UserListItem from "components/UserListItem";
+import { Input, SmallInput } from "components";
+import { updateOrAddPartialAssigment } from "features/expenses/slice";
 
 interface IProps {
   transaction: ICreateTransaction;
   group: ITransactionGroup;
 }
 
-// TODO
+const validInput = new RegExp(
+  '^\\d+(\\.\\d{1,2})?\\.?$'
+);
+
+const SAVE_DELAY_MS = 300;
+
+const UserAmountComponent = (partialsAssignments: IPartialAssignments) => {
+  const dispatch = useDispatch();
+  const saveTimer = useRef<any>();
+
+  const [form, setForm] = useState<IPartialAssignments>(partialsAssignments);
+  const [tmpAmount, setTmpAmount] = useState<string | null>(null);
+
+  const onInputTextChanged = useCallback((_name: string, value: string) => {
+    value = value.replace(',', '.');
+    if (!validInput.test(value) && value !== '') return;
+
+    if (value.endsWith('.')) {
+      setTmpAmount(value);
+      return;
+    }
+    setTmpAmount(null);
+
+    setForm((form) => ({
+      ...form,
+      partialAmount: +value,
+    }));
```

Comment:
With only one `input` you don't need to have `form` object. You can simply have `const [value, setValue] = useState()`. Since you don't need to store the whole `IPartialAssignments` in the form, you can simply use `form.partialAmount` to initialize the `value`. You can use the other data needed for `dispatch(updateOrAddPartialAssigment(form))` from the `props`, it is unnecessary to store it in the `state`.